operational/azure/azure_long_running_instances/azure_long_running_instances.pt

```diff
@@ -0,0 +1,281 @@
+    end
+  end
+end
+
+# Lists All Virtual Machines in each subscription
+datasource "ds_vm_list" do
+  # https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/list-all?tabs=HTTP
+  iterate $ds_subscription_list
+  request do
+    auth $auth_azure
+    pagination $pagination_azure
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/providers/Microsoft.Compute/virtualMachines"])
+    query "api-version","2022-08-01"
+    header "User-Agent", "RS Policies"
+  end
+  result do
+    encoding "json"
+    collect jmes_path(response, "value[*]") do
+      field "id", jmes_path(col_item,"id")
+      field "rg", get(4, split(jmes_path(col_item,"id"), "/"))
+      field "name", jmes_path(col_item,"name")
+      field "subscriptionId",val(iter_item,"subscriptionId")
+      field "subscriptionName",val(iter_item,"displayName")
+    end
+  end
+end
+
+# Retrieves information about the run-time state of each virtual machine
+datasource "ds_vm_instanceview" do
+  # https://learn.microsoft.com/en-us/rest/api/compute/virtual-machines/instance-view?tabs=HTTP
+  iterate $ds_vm_list
+  request do
+    auth $auth_azure
+    pagination $pagination_azure
+    host "management.azure.com"
+    path join(["/subscriptions/", val(iter_item,"subscriptionId"), "/resourceGroups/", val(iter_item,"rg"), "/providers/Microsoft.Compute/virtualMachines/",val(iter_item,"name"),"/instanceView"])
+    query "api-version","2022-08-01"
+    header "User-Agent", "RS Policies"
+    ignore_status [404]
+  end
+  result do
+    encoding "json"
+    field "rg", val(iter_item,"rg")
+    field "name", val(iter_item,"name")
+    field "statuses", jmes_path(response,"statuses")
+    field "subscriptionId",val(iter_item,"subscriptionId")
+    field "subscriptionName",val(iter_item,"subscriptionName")
+  end
+end
+
+# Calls the script to get only the virtual machines that are running
+datasource "ds_vm_running" do
+    run_script $js_filter_vm_running, $ds_vm_instanceview, $param_days_old
+  end
+
+###############################################################################
+# Scripts
+###############################################################################
+
+# Returns instances that are long running
+script "js_filter_vm_running", type: "javascript" do
+    parameters "ds_vm_instanceview", "param_days_old"
+    result "result"
+    code <<-EOS
+    var result=[];
+    _.each(ds_vm_instanceview, function(instance){
+      var codes = _.pluck(instance.statuses, 'code')
+      if ( _.contains(codes, "PowerState/running") ) {
+        var runningState = _.where(instance.statuses, {"code": "PowerState/running"});
+        var displayStatus = runningState[0]["displayStatus"];
+        var provisionState = _.where(instance.statuses, {"code": "ProvisioningState/succeeded"});
+        var timeRunning = provisionState[0]["time"];
+        var timeofevent = new Date(timeRunning);
+        var nowtime = Date.now();
```

Comment:
The date returned by `Date.now()` is UTC, do we need to use a specific timezone for this policy template or is UTC just fine?

Reply:
> The date returned by `Date.now()` is UTC, do we need to use a specific timezone for this policy template or is UTC just fine?

Is the time zone in the azure's response?

Reply:
By default Azure virtual machines use UTC, so it would be better to use UTC for the policy, unless it is known that the machines are set in other time zone, then the policy should use that specific timezone too.

Reply:
That sounds good to me @Rangelmv23 , thank you!